impl/spanner/mutations/table.go

```diff
@@ -0,0 +1,320 @@
+}
+func timeToMark(t time.Time) water.Mark {
+	m := t.Truncate(tsRes)
+	v := uint64(m.UnixNano()) / uint64(tsRes)
+	return water.NewMark(v)
+}
+
+// Table implements a queue of mutations.
+type Table struct {
+	client *spanner.Client
+}
+
+// LogStatusCols are the columns in the LogStatus table.
+type LogStatusCols struct {
+	DirectoryID string
+	LogID       int64
+	WriteToLog  bool
+}
+
+// New returns a new Table object.
+func New(client *spanner.Client) *Table {
+	return &Table{client: client}
+}
+
+// SetWritable enables or disables new writes from going to logID.
+func (t *Table) SetWritable(ctx context.Context, directoryID string, logID int64, enabled bool) error {
+	rtx := t.client.Single()
+	defer rtx.Close()
+
+	m, err := spanner.UpdateStruct(logTable, LogStatusCols{
+		DirectoryID: directoryID,
+		LogID:       logID,
+		WriteToLog:  enabled,
+	})
+	if err != nil {
+		return err
+	}
+	_, err = t.client.Apply(ctx, []*spanner.Mutation{m})
+	return err
+}
+
+// AddLogs adds the logIDs to the list of active logs to send mutations to, and read mutations from.
+// If directoryID and logID already exist, they will be marked as writable.
+func (t *Table) AddLogs(ctx context.Context, directoryID string, logIDs ...int64) error {
+	if len(logIDs) == 0 {
+		return fmt.Errorf("no logs to add")
+	}
+
+	rtx := t.client.Single()
+	defer rtx.Close()
+
+	mutations := make([]*spanner.Mutation, 0, len(logIDs))
+	for _, logID := range logIDs {
+		// Ignore already exists errors so this method can be retried.
+		m, err := spanner.InsertOrUpdateStruct(logTable, LogStatusCols{
+			DirectoryID: directoryID,
+			LogID:       logID,
+			WriteToLog:  true,
+		})
+		if err != nil {
+			return err
+		}
+		mutations = append(mutations, m)
+	}
+	_, err := t.client.Apply(ctx, mutations)
+	return err
+}
+
+// ListLogs returns the mutation logIDs assocciated with directoryID.
+// If writable is true, the list is filtered to only contain writable logs.
+// Returns codes.NotFound if the list is empty.
+func (t *Table) ListLogs(ctx context.Context, directoryID string, writable bool) ([]int64, error) {
+	var stmt spanner.Statement
+	if writable {
+		stmt = spanner.NewStatement(
+			`SELECT LogID FROM LogStatus WHERE DirectoryID = @directory_id AND WriteToLog = TRUE`)
+	} else {
+		stmt = spanner.NewStatement(
+			`SELECT LogID FROM LogStatus WHERE DirectoryID = @directory_id`)
+	}
+	stmt.Params["directory_id"] = directoryID
+	var logIDs []int64
+	rtx := t.client.Single().WithTimestampBound(spanner.MaxStaleness(readStaleness))
+	defer rtx.Close()
+	if err := rtx.Query(ctx, stmt).Do(
+		func(row *spanner.Row) error {
+			var logID int64
+			if err := row.Columns(&logID); err != nil {
+				return err
+			}
+			logIDs = append(logIDs, logID)
+			return nil
+		}); err != nil {
+		return nil, err
+	}
+	if len(logIDs) == 0 {
+		return nil, status.Errorf(codes.NotFound, "not found")
+	}
+	return logIDs, nil
+}
+
+// Send submits an item to the queue and returns the commit timestamp.
```

Comment:
```suggestion
// Send submits a batch of items to the queue and returns the commit timestamp.
```

This concept of a batch is important as it comes up below when returning oversized batches.